### PR TITLE
Upgrade to Lodash v4

### PR DESCRIPTION
**Pull request by nathanstitt**
Continuing the work started at https://github.com/AmpersandJS/ampersand-collection-lodash-mixin/pull/9

Only update needed seems to be that `_.forOwn` (and all the other v4 methods) lacks the `thisArg` for setting the scope.  I worked around that using a closure to this, but could also use `bind` if that'd be preferred.

### CODE REVIEW

File: ampersand-state.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
 'use strict';
 /*$AMPERSAND_VERSION*/
-var uniqueId = require('lodash.uniqueid');
-var assign = require('lodash.assign');
+var uniqueId = require('lodash/uniqueid');
```
Comment (pgilad) [inline ampersand-state.js]: `uniqueId`

**Comment by pgilad** [comment]
You need to camelCase all lodash imports - some OSes are case sensitive.

**Comment by dhritzkiv** [comment]
I believe this PR may be a duplicate of #236, which is waiting on a merge + release

**Comment by nathanstitt** [comment]
Oops, so it is. I didn't notice that one when I created this. We can close this one in favor of #236 since it does more than just the upgrade.

**Pull request closed without merging**